retrieval/targetmanager.go

```diff
@@ -139,11 +142,25 @@ func (tm *TargetManager) stop(removeTargets bool) {
 	log.Info("Stopping target manager...")
 	defer log.Info("Target manager stopped.")
 
-	for _, provs := range tm.providers {
-		for _, p := range provs {
+	tm.m.Lock()
+	provs := []TargetProvider{}
```

Comment:
`tm.providers` is of type `map[*ScrapeConfig][]TargetProvider`
So this will not allocate a meaningful length. Finding out the final length is not worth it here.